pkg/querier/queryrange/codec.go

```diff
@@ -828,16 +828,22 @@ func (Codec) MergeResponse(responses ...queryrangebase.Response) (queryrangebase
 		lokiSeriesRes := responses[0].(*LokiSeriesResponse)
 
 		var lokiSeriesData []logproto.SeriesIdentifier
-		uniqueSeries := make(map[string]struct{})
+		uniqueSeries := make(map[uint64]struct{})
+		//uniqueSeries := make(map[string]struct{})
 
 		// only unique series should be merged
+		b := make([]byte, 0, 1024)
+		keyBuffer := make([]string, 0, 32)
```

Comment:
Maybe this is too tricky. I could create a `Hasher` struct that holds the buffers. It would not be threadsafe though.